### PR TITLE
Add control for kind_of_watermark and reaction time.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/threading_service_config.h [hunk 1]
```diff
@@ -17,21 +18,25 @@ class ThreadingServiceConfig {
...
-    uint32_t    _indexingThreads;
-    uint32_t    _defaultTaskLimit;
-    uint32_t    _semiUnboundTaskLimit;
-    OptimizeFor _optimize;
+    uint32_t           _indexingThreads;
+    uint32_t           _defaultTaskLimit;
+    uint32_t           _semiUnboundTaskLimit;
+    OptimizeFor        _optimize;
+    uint32_t           _kindOfWatermark;
+    vespalib::duration _reactionTime;         // Minimum reaction time to new tasks
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/server/threading_service_config.h:26]: ... or maximum?

**Review by havardpe — APPROVED** [review]
I am a bit worried that this is getting complicated and entangled... Ideally I would prefer separate and specific config settings related to each implementation choice (separate configs might be too heavy, maybe something similar to std::variant as a new config structured type...)

**Comment by baldersheim** [comment]
Agree, but not so worried as I expect these to only have a short lifespan, and will be removed in Q2 after experimentation has converged

File: searchcore/src/vespa/searchcore/proton/server/threading_service_config.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/threading_service_config.h:26]: Yes, min and max are hard....

**Pull request merged**